src/libcollections/str.rs

```diff
@@ -849,10 +854,18 @@ impl str {
     #[unstable(feature = "str_utf16",
                reason = "this functionality may only be provided by libunicode",
                issue = "27714")]
+    #[rustc_deprecated(since = "1.8.0", reason = "renamed to encode_utf16")]
```

Comment:
Argh, this is another one of those cases where we don't really want the deprecation to go into effect on nightly until 1.8 is actually on the *stable* channel. This is probably used rarely enough that it doesn't matter, but we really need to get our story straight there.